### ISSUE TITLE
System.Globalization TimeFormat tests failing for 24-format machine

### ISSUE THREAD

**Issue by Clockwork-Muse** [dotnet/corefx#3454]
My machine is set to display a 24-hr clock, which is causing some of the globalization time-format tests to fail.  The tests are expecting the 12-hr format, but are picking up the format for the local machine instead.

```
System.Globalization.Tests.CultureInfoDateTimeForma.TestShortTimePatternLocale(locale: "en-US", expected: "h:mm tt")
    Assert.Equal() Failure:
        Expected: h:mm tt
        Actual:   HH:mm
````
and

```
System.Globalization.Tests.CultureInfoDateTimeForma.TestLongTimePatternLocale(locale: "en-US", expected: "h:mm:ss tt")
    Assert.Equal() Failure:
        Expected: h:mm:ss tt
        Actual:   HH:mm:ss
````
I'm pretty certain I ran this before, while working on #3020, so it might be a recent change.

**Comment by tarekgh** [comment]
right this test is written in bad way. we shouldn't assume in our tests the date/time formats never change. I'll take care with this issue

**Comment by tarekgh** [comment]
fixed with #3466
